sources/osg/Uniform.js

```diff
@@ -26,395 +27,217 @@ Uniform.isUniform = function ( obj ) {
 
 /** @lends Uniform.prototype */
 Uniform.prototype = {
+
     getName: function () {
-        return this.name;
+        return this._name;
+    },
+
+    dirty: function () {
+        Notify.log( 'deprecated no need anymore' );
     },
+
     getType: function () {
-        return this.type;
+        return this._type;
     },
 
-    get: function () { // call dirty if you update this array outside
-        return this.data;
+    get: function () {
+        Notify.log( 'deprecated use getArray instead' );
+        return this._data;
     },
+
     set: function ( array ) {
-        this.data = array;
-        this.dirty();
-    },
-    dirty: function () {
-        this._dirty = true;
+        Notify.log( 'deprecated use setArray instead' );
+        this._data = array;
     },
+
     apply: function ( gl, location ) {
-        if ( this._dirty ) {
-            this.update.call( this.glData, this.data );
-            this._dirty = false;
-        }
-        this.glCall( gl, location, this.glData );
+        if ( this._isMatrix )
+            gl[ this._glCall ]( location, this._transpose, this._data );
+        else
+            gl[ this._glCall ]( location, this._data );
     },
-    applyMatrix: function ( gl, location ) {
-        if ( this._dirty ) {
-            this.update.call( this.glData, this.data );
-            this._dirty = false;
-        }
-        this.glCall( gl, location, this.transpose, this.glData );
+
+    setArray: function ( array ) {
```

Comment:
I changed to fit the osg api, setArray / getArray set the container on which other functions will copy their data.
On c++ you can override type so in C++ set is used. But because we can't do that in js, I did a set + the type. https://github.com/cedricpinson/osg/blob/master/include/osg/Uniform
I am open to other suggestions that fit osg API